presto-verifier/src/main/java/com/facebook/presto/verifier/framework/AbstractVerification.java

```diff
@@ -205,9 +205,6 @@ public VerificationResult run()
                 QueryBundle controlQueryBundle = control.get();
                 controlQueryFuture = executor.submit(() -> runMainQuery(controlQueryBundle.getQuery(), CONTROL, controlQueryContext));
             }
-            else {
```

Comment:
Instead of two`if and else` can we do everything here like

            // Start control query
            if (!skipControl) {
                QueryBundle controlQueryBundle = control.get();
                controlQueryFuture = executor.submit(() -> runMainQuery(controlQueryBundle.getQuery(), CONTROL, controlQueryContext));
                controlQueryResult = getFutureValue(controlQueryFuture);
                controlQueryContext.setState(QueryState.SUCCEEDED);
            }
            else {
                controlQueryContext.setState(NOT_RUN);
            }

Reply:
That doesn't achieve the objective of running test and control concurrently since it blocks on the control future without having started the test query.

Reply:
Ahh okay . got it .